xchange-coinbasepro/src/main/java/org/knowm/xchange/coinbasepro/service/CoinbaseProAccountServiceRaw.java

```diff
@@ -16,116 +17,178 @@
 import org.knowm.xchange.currency.Currency;
 import org.knowm.xchange.utils.timestamp.UnixTimestampFactory;
 
+import static org.knowm.xchange.coinbasepro.CoinbaseProResilience.PRIVATE_REST_ENDPOINT_RATE_LIMITER;
+
 public class CoinbaseProAccountServiceRaw extends CoinbaseProBaseService {
 
-  public CoinbaseProAccountServiceRaw(Exchange exchange) {
-    super(exchange);
+  public CoinbaseProAccountServiceRaw(
+          CoinbaseProExchange exchange,
+          ResilienceRegistries resilienceRegistries) {
+    super(exchange, resilienceRegistries);
   }
 
   public org.knowm.xchange.coinbasepro.dto.account.CoinbaseProAccount[] getCoinbaseProAccountInfo()
       throws CoinbaseProException, IOException {
-    return coinbasePro.getAccounts(
-        apiKey, digest, UnixTimestampFactory.INSTANCE.createValue(), passphrase);
+    return decorateApiCall(() ->
+            coinbasePro.getAccounts(
+                    apiKey, digest, UnixTimestampFactory.INSTANCE.createValue(), passphrase))
+            .withRateLimiter(rateLimiter(PRIVATE_REST_ENDPOINT_RATE_LIMITER))
+            .call();
   }
 
+  /**
+   * https://docs.pro.coinbase.com/#fees
+   */
   public CoinbaseProFee getCoinbaseProFees() throws CoinbaseProException, IOException {
-    return coinbasePro.getFees(
-        apiKey, digest, UnixTimestampFactory.INSTANCE.createValue(), passphrase);
+    return decorateApiCall(() ->
+            coinbasePro.getFees(
+                    apiKey, digest, UnixTimestampFactory.INSTANCE.createValue(), passphrase))
+            .withRateLimiter(rateLimiter(PRIVATE_REST_ENDPOINT_RATE_LIMITER))
+            .call();
   }
 
   public CoinbaseProSendMoneyResponse sendMoney(
       String accountId, String to, BigDecimal amount, Currency currency)
       throws CoinbaseProException, IOException {
-    return coinbasePro.sendMoney(
-        new CoinbaseProSendMoneyRequest(to, amount, currency.getCurrencyCode()),
-        apiKey,
-        digest,
-        UnixTimestampFactory.INSTANCE.createValue(),
-        passphrase,
-        accountId);
+    return decorateApiCall(() ->
+            coinbasePro.sendMoney(
+                    new CoinbaseProSendMoneyRequest(to, amount, currency.getCurrencyCode()),
+                    apiKey,
+                    digest,
+                    UnixTimestampFactory.INSTANCE.createValue(),
+                    passphrase,
+                    accountId))
+            .withRateLimiter(rateLimiter(PRIVATE_REST_ENDPOINT_RATE_LIMITER))
```

Comment:
if this an idempotent operation?

if two request get to the server because of a retry will it not send money twice?

Reply:
I don't have any retries here? If there is no retry logic why would it send it again it's just a rate limiter?

Reply:
a sorry I misunderstood the code